pkg/collect/rook.go

```diff
@@ -0,0 +1,163 @@
+package collect
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+type CmdResult struct {
+	Stdout []byte
+	Stderr []byte
+}
+
+type RookCmd struct {
+	Cmd      []string
+	Args     []string
+	Filename string // file in the support-bundle
+}
+
+var RookCmds = []RookCmd{
+	RookCmd{
+		Cmd:      []string{"ceph", "status"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "status",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "fs", "status"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "fs",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "fs", "ls"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "fs-ls",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "osd", "status"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "osd-status",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "osd", "tree"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "osd-tree",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "osd", "pool", "ls", "detail"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "osd-pool",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "health", "detail"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "health",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "auth", "ls"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "auth",
+	},
+}
+
+func Rook(ctx *Context) ([]byte, error) {
+	client, err := kubernetes.NewForConfig(ctx.ClientConfig)
+	if err != nil {
+		return nil, errors.New("Failed to create kubernetes clientset")
+	}
+
+	namespace := "rook-ceph"
+	selector := []string{"app=rook-ceph-operator"}
```

Comment:
is this a kURL label or a standard label that all rook pods will have?

Reply:
kURL label

Reply:
See latest commits.  I just added the ability for the user to specify rook's namespace, selector, and/or container names in the Rook Collector spec (handles the case where user ran KOTS install and manually had installed cephs).  If not specified in the Collector spec, then we'll default to these.  I also added a comment so its clear that kURL labeled this pod.